exporter/splunkhecexporter/tracedata_to_splunk.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecexporter
+
+import (
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.uber.org/zap"
+)
+
+type splunkEvent struct {
+	Time       int64       `json:"time"`                 // epoch time
+	Host       string      `json:"host"`                 // hostname
+	Source     string      `json:"source,omitempty"`     // optional description of the source of the event; typically the app's name
+	SourceType string      `json:"sourcetype,omitempty"` // optional name of a Splunk parsing configuration; this is usually inferred by Splunk
+	Index      string      `json:"index,omitempty"`      // optional name of the Splunk index to store the event in; not required if the token has a default index set in Splunk
+	Event      interface{} `json:"event"`                // Payload of the event.
+}
+
+func traceDataToSplunk(logger *zap.Logger, data consumerdata.TraceData, config *Config) ([]*splunkEvent, int, error) {
+	var host string
+	if data.Resource != nil {
+		host = data.Resource.Labels[hostnameLabel]
+	}
+	if host == "" {
+		host = unknownHostName
+	}
+	numDroppedSpans := 0
+	splunkEvents := make([]*splunkEvent, 0)
+	for _, span := range data.Spans {
+		if span.StartTime == nil {
+			logger.Warn(
+				"Span dropped as it had no start timestamp",
+				zap.Any("span", span))
+			numDroppedSpans++
+			continue
+		}
+		se := &splunkEvent{
+			Time:       timestampToEpochMilliseconds(span.StartTime),
+			Host:       host,
+			Source:     config.Source,
+			SourceType: config.SourceType,
+			Index:      config.Index,
+			Event:      span,
```

Comment:
What assurances are there that this won't shift with unrelated internal type changes?  Seems less safe than the current metric translator: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/82b1adf3e12fdee23a2ea28c3c14b2c08a2ed477/exporter/splunkhecexporter/metricdata_to_splunk.go#L77

Reply:
I don't have any assurances, and I have superficial knowledge of HEC. I read the docs, used it, and it works reasonably well for me.

I would however point out that Splunk is in GA using this format, and the format has been used since the inception of HEC. If anything I would expect Splunk to change its HEC input in a backwards-compatible manner and with long deprecation periods.

What assurances do you require? How can I help?

Reply:
Sry, I'm referring specifically to the json encoding of the `tracepb.Span`s as is and not the overall HEC event format, which I have not worked with before.  It's unclear to me that the spans are being marshaled as expected without using jsonpb/protojson directly.

Reply:
That's a good point. I'll add tests for that. Maybe that will help coverage enough too :)